receiver/splunkenterprisereceiver/client.go

```diff
@@ -167,7 +158,16 @@ func (c *splunkEntClient) makeRequest(req *http.Request) (*http.Response, error)
 	if eptType == nil {
 		return nil, errCtxMissingEndpointType
 	}
-	if sc, ok := c.clients[eptType]; ok {
+
+	var endpointType string
+	switch t := eptType.(type) {
+	case string:
+		endpointType = t
+	default:
+		endpointType = fmt.Sprintf("%v", eptType)
```

Comment:
You can call `fmt.Sprint` here instead of providing the formatter

Reply:
```suggestion
		endpointType = fmt.Sprint(eptType)
```
like this?

Reply:
that wont work without a type assertion because eptType is any

Reply:
in the default case